### PR TITLE
lodestar-cli: fix init fields copy bug

### LINKED ISSUE #1266: --testnet command copies over extra fields
**Describe the bug**

When running `yarn run cli beacon init --testnet altona`, the resultant beacon.config.json also contains these parameters:
```
  "templateConfigFile": null,
  "testnet": "altona"
```

**Expected behavior**

I would expect to see the config created without those 2 parameters mentioned above.

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**
Fixes #1266 to copy over the necessary fields to the beacon.config.json when running `yarn run cli beacon init --testnet altona`

### CODE REVIEW

File: packages/lodestar-cli/src/options/beaconNodeOptions/api.ts [hunk 1]
```diff
@@ -2,37 +2,37 @@ import {Options} from "yargs";
...
+    alias: ["api.host", "api.rest.host"],
     type: "string",
     defaultDescription: defaultOptions.api.rest.host,
     group: "api",
   } as Options,
 
-  "api.rest.port": {
-    alias: ["api.port"],
+  "api.port": {
+    alias: ["api.port", "api.rest.port"],
```
Comment (dapplion) [inline packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: The property names here cannot be changed, they must match the actual BeaconNodeOption types.

File: packages/lodestar-cli/src/options/beaconNodeOptions/network.ts [hunk 2]
```diff
@@ -27,13 +27,13 @@ export const networkOptions = {
...
 
-  "network.bootnodes": {
+  "network.bootMultiaddrs": {
     type: "array",
     defaultDescription: JSON.stringify(defaultOptions.network.bootnodes),
     group: "network",
   } as Options,
 
-  "network.multiaddrs": {
+  "network.localMultiaddrs": {
```
Comment (dapplion) [inline packages/lodestar-cli/src/options/beaconNodeOptions/network.ts]: The property names here cannot be changed, they must match the actual BeaconNodeOption types.

**Review by dapplion — CHANGES_REQUESTED** [review]

File: packages/lodestar-cli/src/options/beaconNodeOptions/api.ts (hunk 1 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: so does that mean we need to restructure our altona configs going forward?  for example, right now in `testnets/altona.ts`, the structure for the `api` options are as follows:

```
  "api": {
    "enabled": true,
    "namespaces": ["beacon", "validator"],
    "host": "127.0.0.1",
    "port": 9596,
    "cors": {
      "origin": "*"
    }
  },
```

Do you mean to say that now it should look something like this instead?
```
 "api": {
    "rest": {
      "enabled": true,
      "namespaces": ["beacon", "validator"],
      "host": "127.0.0.1",
      "port": 9596,
      "cors": {
        "origin": "*"
      }
    }
  },
```

I'm not sure why we need the `.rest` part.  I thought that was just an alias for when we were still using the old `mergeBeaconOptions` method?

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: To clarify, there aren't any other subcategories of `api` options, so I don't see a reason to subcategorize them with `rest`

Comment (wemeetagain) [reply packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: @3xtr4t3rr3str14l 
We've simplified the configuration to _require_ one-for-one matching of yargs `Options` to `BeaconNodeOptions`. (if there's a configuration file that doesn't match the structure of `BeaconNodeOptions` (with a few additional fields, eg: `genesisStateFile`), you can think of that as a bug)
I think what you're hoping for is a matter of renaming the options in `BeaconNodeOptions`. (in the same vein as https://github.com/ChainSafe/lodestar/issues/1226) and pushing those changes to the yargs level as well. Might add a comment there, as there are quite a few fields in `BeaconNodeOptions` that could/should be renamed.

File: packages/lodestar-cli/src/options/beaconNodeOptions/network.ts (hunk 2 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/options/beaconNodeOptions/network.ts]: changes for this file reverted

File: packages/lodestar-cli/src/options/beaconNodeOptions/api.ts (hunk 1 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: changes for this file reverted

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/options/beaconNodeOptions/api.ts]: also, made my comment in #1226 accordingly to request to remove the `api` nest

**Comment by wemeetagain** [comment]
Should be resolved with #1301

**Pull request closed without merging**